pkg/flagset/flagset.go

```diff
@@ -56,159 +58,158 @@ func ServerWithConfig(cfg *config.Config) []cli.Flag {
-			EnvVar:      "PHOENIX_DEBUG_TOKEN",
+			EnvVars:     []string{"PHOENIX_DEBUG_TOKEN"},
 			Destination: &cfg.Debug.Token,
 		},
 		&cli.BoolFlag{
 			Name:        "debug-pprof",
 			Usage:       "Enable pprof debugging",
-			EnvVar:      "PHOENIX_DEBUG_PPROF",
+			EnvVars:     []string{"PHOENIX_DEBUG_PPROF"},
 			Destination: &cfg.Debug.Pprof,
 		},
 		&cli.BoolFlag{
 			Name:        "debug-zpages",
 			Usage:       "Enable zpages debugging",
-			EnvVar:      "PHOENIX_DEBUG_ZPAGES",
+			EnvVars:     []string{"PHOENIX_DEBUG_ZPAGES"},
 			Destination: &cfg.Debug.Zpages,
 		},
 		&cli.StringFlag{
 			Name:        "http-addr",
 			Value:       "0.0.0.0:9100",
 			Usage:       "Address to bind http server",
-			EnvVar:      "PHOENIX_HTTP_ADDR",
+			EnvVars:     []string{"PHOENIX_HTTP_ADDR"},
 			Destination: &cfg.HTTP.Addr,
 		},
 		&cli.StringFlag{
 			Name:        "http-root",
 			Value:       "/",
 			Usage:       "Root path of http server",
-			EnvVar:      "PHOENIX_HTTP_ROOT",
+			EnvVars:     []string{"PHOENIX_HTTP_ROOT"},
 			Destination: &cfg.HTTP.Root,
 		},
 		&cli.StringFlag{
 			Name:        "http-namespace",
 			Value:       "com.owncloud.web",
 			Usage:       "Set the base namespace for the http namespace",
-			EnvVar:      "PHOENIX_NAMESPACE",
+			EnvVars:     []string{"PHOENIX_NAMESPACE"},
 			Destination: &cfg.HTTP.Namespace,
 		},
 		&cli.StringFlag{
 			Name:        "asset-path",
 			Value:       "",
 			Usage:       "Path to custom assets",
-			EnvVar:      "PHOENIX_ASSET_PATH",
+			EnvVars:     []string{"PHOENIX_ASSET_PATH"},
 			Destination: &cfg.Asset.Path,
 		},
 		&cli.StringFlag{
 			Name:        "web-config",
 			Value:       "",
 			Usage:       "Path to phoenix config",
-			EnvVar:      "PHOENIX_WEB_CONFIG",
+			EnvVars:     []string{"PHOENIX_WEB_CONFIG"},
 			Destination: &cfg.Phoenix.Path,
 		},
 		&cli.StringFlag{
 			Name:        "web-config-server",
 			Value:       "http://localhost:9140",
 			Usage:       "Server URL",
-			EnvVar:      "PHOENIX_WEB_CONFIG_SERVER",
+			EnvVars:     []string{"PHOENIX_WEB_CONFIG_SERVER"},
 			Destination: &cfg.Phoenix.Config.Server,
 		},
 		&cli.StringFlag{
 			Name:        "web-config-theme",
 			Value:       "owncloud",
 			Usage:       "Theme",
-			EnvVar:      "PHOENIX_WEB_CONFIG_THEME",
+			EnvVars:     []string{"PHOENIX_WEB_CONFIG_THEME"},
 			Destination: &cfg.Phoenix.Config.Theme,
 		},
 		&cli.StringFlag{
 			Name:        "web-config-version",
 			Value:       "0.1.0",
 			Usage:       "Version",
-			EnvVar:      "PHOENIX_WEB_CONFIG_VERSION",
+			EnvVars:     []string{"PHOENIX_WEB_CONFIG_VERSION"},
 			Destination: &cfg.Phoenix.Config.Version,
 		},
-		&cli.StringFlag{
-			Name:   "web-config-apps",
-			Value:  "files,draw-io,pdf-viewer,markdown-editor,media-viewer",
-			Usage:  `String with comma separated values. --web-config-apps "pdf-viewer, files, draw-io"`,
-			EnvVar: "PHOENIX_WEB_CONFIG_APPS",
+		&cli.StringSliceFlag{
+			Name:    "web-config-apps",
```

Comment:
Maybe name it in singular? That would show it a little bit more that you got to provide this flag per app.

Reply:
agree